src/MessageSource.php

```diff
@@ -31,13 +32,16 @@ public function __construct(
      */
     public function getMessage(string $id, string $category, string $locale, array $parameters = []): ?string
     {
-        $this->bindDomain($category);
-        $this->setLocale($locale);
-        $n = current($parameters);
-        if (is_int($n) === false) {
-            return dgettext($category, $id);
+        $result = $this->getMessageInternal($id, $category, $locale, $parameters);
+        if (!$this->strictMode) {
+            return $result;
```

Comment:
Non-strict mode is bug. See https://github.com/yiisoft/translator/blob/master/src/MessageReaderInterface.php:

```php
  /**
   * Obtains a translation message.
   *
   * @param string $id ID of the message to get.
   * @param string $category Category of the message to get.
   * @param string $locale Locale of the message to get.
   * @param array $parameters Parameters that may be used to decide about the message to obtain.
   *
   * @return string|null The translated message or null if translation wasn't found.
   */
  public function getMessage(string $id, string $category, string $locale, array $parameters = []): ?string;
```

Non-strict mode violates it: `The translated message or null if translation wasn't found`

Reply:
https://www.php.net/manual/en/function.gettext.php
```
gettext([string](https://www.php.net/manual/ru/language.types.string.php) $message): [string](https://www.php.net/manual/ru/language.types.string.php)
```

> Returns a translated [string](https://www.php.net/manual/en/language.types.string.php) if one is found in the translation table, or the submitted message if not found.